### PR TITLE
Rename stream tables to blink tables

### LINKED ISSUE #3694: Rename "stream table" to "blink table"
It's been determined that "stream table" is a potentially confusing name now that we have begun to call our tables "streaming tables" generally. "Blink table" is the new consensus name. This requires updating the attribute name, accessor names, method names, Kafka ingest code in Java and Python, and ensuring a documentation task is addressed in time for month end.

### PR DESCRIPTION
**Pull request by rcaudy**
Also improve documentation and code formatting in a few related places.
Fixes #3694 
See https://github.com/deephaven/deephaven-core/issues/3694#issuecomment-1555198245 for a summary of required documentation updates.

### CODE REVIEW

**Comment by rcaudy** [comment]
Nightlies are passing, but for a random-looking test executor failure for the Java 11 testOutOfBand (https://github.com/rcaudy/deephaven-core/actions/runs/5035201987/jobs/9030554996).

File: engine/table/src/main/java/io/deephaven/stream/StreamToBlinkTableAdapter.java [hunk 1]
```diff
@@ -40,14 +40,15 @@
...
 
 /**
- * Adapter for converting streams of data into columnar Deephaven {@link Table tables}.
+ * Adapter for converting streams of data into columnar Deephaven {@link Table tables} that conform to
+ * {@link Table#BLINK_TABLE_ATTRIBUTE blink table} semantics.
  *
  * @implNote The constructor publishes {@code this} to the {@link UpdateGraphProcessor} and thus cannot be subclassed.
  */
-public class StreamToTableAdapter extends ReferenceCountedLivenessNode
+public class StreamToBlinkTableAdapter extends ReferenceCountedLivenessNode
```
Comment (devinrsmith) [inline engine/table/src/main/java/io/deephaven/stream/StreamToBlinkTableAdapter.java:48]: Good non-rename  of stream.

File: py/server/deephaven/stream/__init__.py [hunk 2]
```diff
@@ -9,14 +9,14 @@
 from deephaven import DHError
 from deephaven.table import Table
 
-_JStreamTableTools = jpy.get_type("io.deephaven.engine.table.impl.StreamTableTools")
+_JBlinkTableTools = jpy.get_type("io.deephaven.engine.table.impl.BlinkTableTools")
 
 
 def stream_to_append_only(table: Table) -> Table:
```
Comment (devinrsmith) [inline py/server/deephaven/stream/__init__.py:15]: Do we want to rename this method? Create a new method, and deprecate this one?

Comment (rcaudy) [reply py/server/deephaven/stream/__init__.py:15]: Definitely rename. I'm a little iffy on creating a new method and deprecating, but I think doing so would be in-line with my current deprecation strategy.

**Review by jmao-denver — APPROVED** [review]

**Review by devinrsmith — APPROVED** [review]

**Comment by deephaven-internal** [comment]
Labels indicate documentation is required. Issues for documentation have been opened:

How-to: https://github.com/deephaven/deephaven.io/issues/2623
Conceptual: https://github.com/deephaven/deephaven.io/issues/2621
Reference: https://github.com/deephaven/deephaven.io/issues/2620
Blog: https://github.com/deephaven/deephaven.io/issues/2622

**Pull request merged**